File: All Languages/Java Codes/QuickSort.java

// Java program to demonstrate working of Arrays.toString()
import java.util.*;

class QuickSort{
    static void Quick(int arr[],int low,int high){
        if (low>=high) {
            return;
        }
        int start=low;
        int end=high;
        int mid=(start)+(end-start)/2;
        int pivot=arr[mid];
        while (start<=end) {
            while (arr[start]<pivot) {
                start++;
            }
            while (arr[end]>pivot) {
                end--;
            }
            if(start<=end){
                int temp=arr[start];
                arr[start]=arr[end];
                arr[end]=temp;
                start++;
                end--;
                
            }
        }
        Quick(arr,low,end);
        Quick(arr,start,high);
    }
    public static void main(String[] args) {
       int arr[]=new int[10];
       for (int i = 0; i < arr.length; i++) {
        arr[i]=10-i;
       }
       Quick(arr,0,9);
       System.out.println(Arrays.toString(arr));
    }

    
}

File: All Languages/Rust Codes/quick_sort.rs


// quick sort
fn quick_sort(arr: &mut [i32]) {
    let len = arr.len();
    if len < 2 {
        return;
    }
    let pivot = arr[len / 2];
    let mut i = 0;
    let mut j = len - 1;
    while i <= j {
        while arr[i] < pivot {
            i += 1;
        }
        while arr[j] > pivot {
            j -= 1;
        }
        if i <= j {
            arr.swap(i, j);
            i += 1;
            j -= 1;
        }
    }
    if 0 < j {
        quick_sort(&mut arr[0..=j]);
    }
    if i < len - 1 {
        quick_sort(&mut arr[i..]);
    }
}

fn main() {
    let mut arr = [4, 2, 5, 3, 1];
    quick_sort(&mut arr);
    println!("{:?}", arr);
}
```
